app/services/cosmos/cosmos.ts

```diff
@@ -113,15 +185,36 @@ export class CosmosAPI {
    * @param delegatorAddress The delegator address
    * @param validatorAddress The validator address
    */
-  async getUnbondingDelegation(delegatorAddress: string, validatorAddress: string) {
-    return this.api.get.unbondingDelegation(delegatorAddress, validatorAddress) as CosmosUnbondingDelegation
+  async getUnbondingDelegation(delegatorAddress: string, validatorAddress: string):
+    Promise<CosmosUnbondingDelegation> {
+    const { unbond } =
+      await this.queryClient.staking.unbondingDelegation(delegatorAddress, validatorAddress)
+    return convertUnbondingDelegation(unbond)
   }
 
   /**
    * Query the annual provisioned tokens
    */
   async queryAnnualProvision(): Promise<string> {
-    return this.api.get.annualProvisionedTokens()
+    const { annualProvisions } = await this.queryClient.mint.annualProvisions()
+    const provision = new TextDecoder().decode(annualProvisions);
+    return new BigNumber(provision).shiftedBy(18).toFixed();
+  }
+
+  /**
+   * Create a signing client implements the CosmosSigner interface
+   */
+  async createSigningClient(signer: OfflineDirectSigner): Promise<CosmosSigner> {
+    const signingStargateClient = await SigningStargateClient.connectWithSigner(this.restURL, signer)
+    return {
+      async signAndBroadcast(message: CosmosMessageToSign): Promise<BroadcastTxResponse> {
+        const { signerAddress, msgs, fee, memo } = message
+        const result = await signingStargateClient.signAndBroadcast(signerAddress, msgs, fee, memo)
+        // TODO: Could check if broadcast success here
+        // and return Promise<CosmosTxQueryResult> rather than Promise<BroadcastTxResponse>
```

Comment:
Isn't that we intentionally do not check here so that we can handle in the caller?

Reply:
I was thinking if I could wrap all the CosmJS types (like `BroadcastTxResponse`) in our cosmos service when writing this comment.
But it seems not an issue now, never mind.